### PR TITLE
Fix run_go_tests_expanding_packages argument parsing

### LINKED ISSUE #21033: Robustness periodic tests are no longer running the test
### Bug report criteria

- [ ] This bug report is not security related, security issues should be disclosed privately via security@etcd.io.
- [ ] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [ ] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [ ] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

We noticed during regular meeting that https://testgrid.k8s.io/sig-etcd-robustness#ci-etcd-robustness-main-amd64 is empty. Meaning it should include Test scenarios from JUNIT report, but there is none.

The test itself now exits immidietly and whole job runs only for 2 minutes https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-etcd-robustness-main-amd64/2001368803501412352

This has been happening for last 2 weeks.

At least we can say that all the test pass :P

/cc @ahrtr @ivanvc @fuweid 

### What did you expect to happen?

Robustness tests to be running, for at least 1 hours and generate JUNIT report.

### How can we reproduce it (as minimally and precisely as possible)?

https://testgrid.k8s.io/sig-etcd-robustness#ci-etcd-robustness-main-amd64

### Anything else we need to know?

_No response_

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
# paste output here

$ etcdctl version
# paste output here
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

### Relevant log output

```Shell

```

### PR DESCRIPTION
**Pull request by ivanvc**
Escape the regular expression and ensure you capture go.etcd.io packages as well. As defined in `run_go_tests`.

Fixes #21033

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: scripts/test_lib.sh [hunk 1]
```diff
@@ -354,7 +354,7 @@ function run_go_tests_expanding_packages {
   local packages=()
   local args=()
   for arg in "$@"; do
-    if [[ "${arg}" =~ ./ ]]; then
+    if [[ "${arg}" =~ ^\./ || "${arg}" =~ ^go\.etcd\.io/etcd ]]; then
```
Comment (fuweid) [inline scripts/test_lib.sh:357]:
```
➜  etcd git:(main) ag run_go_tests_expanding_packages
scripts/test_lib.sh
349:# run_go_tests_expanding_packages [arguments to pass to go test]
353:function run_go_tests_expanding_packages {

scripts/test.sh
140:      run_go_tests_expanding_packages ./tests/integration/v2store/... \
172:    run_go_tests_expanding_packages ./tests/e2e/... \
177:    run_go_tests_expanding_packages ./tests/common/... \
187:    run_go_tests_expanding_packages ./tests/robustness/... \
221:  run_go_tests_expanding_packages ./tests/integration/... \
230:  run_go_tests_expanding_packages ./tests/e2e/... \
```

**Review by fuweid — APPROVED** [review]

File: scripts/test_lib.sh (hunk 1 above)
Comment (fuweid) [reply scripts/test_lib.sh:357]: Maybe we can run command first and check if it fails.

```
  local golist_output
  if ! golist_output=$(go list "${packages[@]}" 2>&1); then
    log_error -e "failed to go list ${packages[@]}"
    log_error -e "${golist_output}"
    return 2
  fi

  # Expanding patterns (like ./...) into list of packages
  local unpacked_packages=()
  while IFS='' read -r line; do unpacked_packages+=("$line"); done <<< ${golist_output}

```

So, if there is any mismatch, we should panic in CI and catch that error before merge.
